rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/RascalTextDocumentService.java

```diff
@@ -382,7 +385,44 @@ public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
                 ITree cursorTree = findQualifiedNameUnderCursor(focus);
                 return rascalServices.getRename(TreeAdapter.getLocation(cursorTree), focus, workspaceFolders, facts::getPathConfig, params.getNewName()).get();
             })
-            .thenApply(t -> DocumentChanges.translateDocumentChanges(this, t));
+            .thenApply(t -> {
+                showMessages((ISet) t.get(1));
+                return DocumentChanges.translateDocumentChanges(this, (IList) t.get(0));
+            });
+    }
+
+    private <T extends ICollection<T>> void showMessages(ICollection<T> messages) {
```

Comment:
shouldn't it be `T extends IConstructor`?

Reply:
This is the confusing interface of `ISet`. I'll just write `ISet` instead and leave this to the future developer who wants to map an `IList` of messages as well.